contracts/sentinel/guards/CurveSwapperGuard.sol

```diff
@@ -0,0 +1,83 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity ^0.8.4;
+
+import "../IGuard.sol";
+import "../../refs/CoreRef.sol";
+import "../../pcv/PCVGuardian.sol";
+import "../../pcv/curve/ICurvePool.sol";
+import "../../pcv/curve/CurveSwapper.sol";
+
+/// @notice Guard that can trigger a Curve swapper under certain conditions.
+contract CurveSwapperGuard is IGuard, CoreRef {
+    /// @notice the PCV mover contract exposed to guardian role
+    address public immutable pcvGuardian;
+    /// @notice the Curve swapper to call
+    address public immutable curveSwapper;
+    /// @notice the source deposit to withdraw from
+    address public immutable sourceDeposit;
+    /// @notice the PCV Deposit to read balance from
+    address public immutable inspectedDeposit;
+    /// @notice the minimum balance under which a swap can be triggered
+    uint256 public immutable inspectedDepositMinBalance;
```

Comment:
Why is there a concern about only being able to perform this swap once another PCV deposit is running low? Surely we want to swap all LUSD for DAI and so it doesn't matter what the balance of the Compound DAI PCV deposit is for example?

Reply:
Can be discussed but I thought the spec was "holding LUSD is fine, but we want to be able to swap LUSD to DAI if DAI reserves are running low, to defend the peg". Can totally change that

Reply:
see comment below